samples/embedding_precedence.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import "fmt"
+
+type Pet struct {
+	Name string
+}
+
+type Cat struct {
+	Pet
+	Name string
+}
+
+type Animal struct {
+	Name string
+}
+
+type Dog struct {
+	Animal
+	Pet
+}
+
+func main() {
+	pet := Pet{
+		Name: "Charlotte",
+	}
+	cat := Cat{
+		Pet:  pet,
+		Name: "Charly",
+	}
+	fmt.Printf("pet.Name:  %s\n", pet.Name)
+	fmt.Printf("cat.Name:  %s \n\n", cat.Name)
```

Comment:
a very very minor nit: Some of your strings have a space before the `\n` and others dont. I can't see a reason for it, so can they be made consistent? (usually you would not have the space - if the message is as wide as the user's terminal, the space may cause an unnecessary line wrap, adding phantom blank lines.)

Reply:
Sure, I have updated that now.

Reply:
I'm being really pedantic here, but now the code samples do not match what is linked to in the playground - the playground still has the extra spaces. It may seem trivial, but the way I verify these match is to paste the version in this PR into the playground, hit "Share" and verify the same link is generated. If not, I know there are differences. So it is best that they all match.

Here are the links for the versions in this PR:
Embedded Structs: https://goplay.space/#lAl70x0QACE
Embedded Interfaces: https://goplay.space/#vLgi3SMtLZp
Precedence: https://goplay.space/#cmWFJ26jm2E

Can you update the links in the slides to point to these?

Reply:
That's updated now, I did forget to sync the two.